api/src/lib.rs

```diff
@@ -1214,22 +1294,81 @@ impl Client {
         Ok(http_response)
     }
 
-    fn request<LocationT, RequestT, SuccessT, QueryT>(
+    fn splitable_request<LocationT, RequestT, SuccessT, QueryT>(
         &self,
         method: Method,
         url: LocationT,
-        body: Option<RequestT>,
+        body: RequestT,
         query: Option<QueryT>,
         retry: Retry,
+    ) -> Result<SplitableRequestResponse<SuccessT>>
+    where
+        LocationT: IntoUrl + Display + Clone,
+        RequestT: Serialize + SplittableRequest + Clone + Debug,
+        QueryT: Serialize + Clone,
+        for<'de> SuccessT: Deserialize<'de> + ReducibleResponse + Clone + Debug + Default,
+    {
+        debug!("Attempting {method} `{url}`");
+        let result: Result<SuccessT> =
+            self.request(&method, &url, &Some(body.clone()), &query, &retry);
+
+        let should_split = |error: &Error| {
+            if let Error::Api { status_code, .. } = error {
+                *status_code == reqwest::StatusCode::UNPROCESSABLE_ENTITY
+                    || *status_code == reqwest::StatusCode::BAD_REQUEST
+            } else {
+                false
+            }
+        };
+
+        match result {
+            Ok(response) => Ok(SplitableRequestResponse {
+                response,
+                num_failed: 0,
+            }),
+            Err(error) if should_split(&error) => {
+                let mut num_failed = 0;
+
+                let response = SuccessT::reduce(
+                    body.split()
+                        .iter()
```

Comment:
This is looking much better, but we should be able to do better still as this is a standard fold pattern, if you switch back to the `merge` trait method:
```rust
let response = body
    .split()
    .iter()
    .filter_map(|request| {
        match self.request(&method, &url, &Some(request), &query, &retry) {
            Ok(response) => Some(response),
            Err(_) => {
                num_failed += 1;
                None
            }
        }
    })
    .fold(SuccessT::empty(), |merged, next: SuccessT| merged.merge(next));
```